main_test.go

```diff
@@ -45,6 +47,12 @@ func TestMain(t *testing.M) {
 		getMetaDataDao = getMetaDataDaoWithTenant
 		getRhcConnectionDao = getDefaultRhcConnectionDao
 		getApplicationAuthenticationDao = getApplicationAuthenticationDaoWithTenant
+		getAuthenticationDao = getAuthenticationDaoWithTenant
+
+		dao.Vault = &mocks.MockVault{}
```

Comment:
If its an integration test Vault shouldn't be mocked, should be?

Reply:
Well my idea was, that before my change there was only 1 test where we used this MockVault and it is test for status listener. So I wanted to do same thing in auth tests. There was option to add this mock into every particular test which needs Vault or move it in main_test.go.

Reply:
You are right, yes. Let's keep it this way then.

---

What I was saying —which I really didn't notice even though I've been working on it—, is that we're mocking Vault on integration tests, which I guess we shouldn't be doing since they're integration tests. Since this isn't the purpose of this PR I've opened a ticket so we can discuss it there: https://issues.redhat.com/browse/RHCLOUD-18285

Reply:
Yes, that makes sense. Let's discuss.